### PR TITLE
MGDOBR-1220: Bump knative 1.5

### PR DESCRIPTION
**Pull request by r00ta**
[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by manstis** [comment]
Should you add a (temporary, or non-invasive) change to a `pom.xml` on this too; to confirm GHA's are OK?

**Comment by danielezonca** [comment]
In the original PR from @matzew there are also other small changes on an annotation.
See 
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1090/files#diff-282358aebc55f845bd533b6c3aa89c5aea1d011ce8aae3fcd6293dc9eb2908f7 
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1090/files#diff-dbeb6861d2c2c1812b0722a98b2b494a2ada725a8c300724d3f5b1edca1b1d70
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1090/files#diff-a8ccce23b8ed8e5b4a92c012fb2f5d0f51d66764542bf0c5e245a99a2c17ee95

As far as I know it is only deprecated so it should work also without that change

**Comment by sutaakar** [comment]
`error: from storage-version-migration-eventing-: cannot use generate name with apply`

**Comment by r00ta** [comment]
> In the original PR from @matzew there are also other small changes on an annotation. See
> 
> * https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1090/files#diff-282358aebc55f845bd533b6c3aa89c5aea1d011ce8aae3fcd6293dc9eb2908f7
> * https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1090/files#diff-dbeb6861d2c2c1812b0722a98b2b494a2ada725a8c300724d3f5b1edca1b1d70
> * https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1090/files#diff-a8ccce23b8ed8e5b4a92c012fb2f5d0f51d66764542bf0c5e245a99a2c17ee95
> 
> As far as I know it is only deprecated so it should work also without that change

We already migrated to the new annotations

File: dev/bin/knative-installer.sh [hunk 1]
```diff
@@ -15,26 +15,21 @@ else
...
-kubectl apply --filename $eventing_core_url
+kubectl apply -f https://raw.githubusercontent.com/openshift/knative-eventing/release-v1.4/openshift/release/artifacts/eventing-core.yaml
+kubectl create -f https://raw.githubusercontent.com/openshift/knative-eventing/release-v1.4/openshift/release/artifacts/eventing-post-install.yaml
 
 header_text "Waiting for Knative Eventing Core to become ready"
 kubectl wait deployment --all --timeout=900s --for=condition=Available -n knative-eventing
 
 header_text "Initializing Knative Eventing Kafka APIs"
-kubectl apply --filename $eventing_kafka_url
+kubectl apply -f https://raw.githubusercontent.com/openshift-knative/eventing-kafka-broker/release-v1.4/openshift/release/artifacts/eventing-kafka.yaml
```
Comment (robpblake) [inline dev/bin/knative-installer.sh:32]: @r00ta It looks like the `securityContext` definitions in this file are preventing the pods from booting on vanilla k8s. In particular the `runAsNonRoot:true` causes it to fail on my local.

Going to the upstream https://github.com/knative-sandbox/eventing-kafka-broker/releases/download/knative-v1.4.3/eventing-kafka.yaml for 1.4.3 of eventing-kafka, pods all boot as expected

**Comment by r00ta** [comment]
trying to install 1.5 according to the official docs.. https://knative.dev/v1.5-docs/install/

**Review by manstis — APPROVED** [review]

**Comment by sutaakar** [comment]
Upgrade test fail is expected - as the CI tries to install dependencies using scripts from main branch.

**Review by danielezonca — APPROVED** [review]

**Pull request merged**